pandas/tseries/period.py

```diff
@@ -478,7 +485,7 @@ def asfreq(self, freq=None, how='E'):
         """
         how = _validate_end_alias(how)
 
-        freq = frequencies.get_standard_freq(freq)
+        freq = Period._maybe_convert_freq(freq).freqstr
```

Comment:
Looks better to pass freq as it is, because ``_gfc`` can accept offset?

- https://github.com/pydata/pandas/blob/1f55e919a3b85cd0891a7a7a1d6913ca06f6bd9e/pandas/tseries/frequencies.py#L259